Gem/Code/Source/Automation/ScriptReporter.cpp

```diff
@@ -1081,4 +1091,74 @@ namespace AtomSampleViewer
+            totalScreenshotWarningCount += scriptReport.m_screenshotWarningCount;
+
+            AZStd::string assertLogLine = AZStd::string::format("Asserts: %i \n", totalAssertCount);
+            AZStd::string errorsLogLine = AZStd::string::format("Errors: %i \n", totalErrorCount);
+            AZStd::string warningsLogLine = AZStd::string::format("Warnings: %i \n", totalWarningCount);
+            AZStd::string screenshotErrorsLogLine = AZStd::string::format("Screenshot errors: %i \n", totalScreenshotErrorCount);
+            AZStd::string screenshotWarningsLogLine = AZStd::string::format("Screenshot warnings: %i \n", totalScreenshotWarningCount);
+            AZStd::string failedScreenshotsLogLine = "\nFailed screenshot test info below.\n";
+
+            auto io = AZ::IO::LocalFileIO::GetInstance();
+            AZStd::string projectPath = AZ::Utils::GetProjectPath();
+            AZStd::string exportTestResultsFolder;
+            AZStd::string exportFileName = "exportedTestResults.txt";
+            AzFramework::StringFunc::Path::Join(projectPath.c_str(), "TestResults/", exportTestResultsFolder);
+            AzFramework::StringFunc::Path::Join(exportTestResultsFolder.c_str(), "exportedTestResults.txt", exportFileName);
+            AZ::IO::HandleType logHandle;
+
+            io->CreatePath(exportTestResultsFolder.c_str());
+            if (io->Open(exportFileName.c_str(), AZ::IO::OpenMode::ModeWrite, logHandle))
+            {
+                io->Write(logHandle, assertLogLine.c_str(), assertLogLine.size());
+                io->Write(logHandle, errorsLogLine.c_str(), errorsLogLine.size());
+                io->Write(logHandle, warningsLogLine.c_str(), warningsLogLine.size());
+                io->Write(logHandle, screenshotErrorsLogLine.c_str(), screenshotErrorsLogLine.size());
+                io->Write(logHandle, screenshotWarningsLogLine.c_str(), screenshotWarningsLogLine.size());
+                io->Write(logHandle, failedScreenshotsLogLine.c_str(), failedScreenshotsLogLine.size());
+
+                for (const ScreenshotTestInfo& screenshotTest : scriptReport.m_screenshotTests)
+                {
+                    AZStd::string baselineImagePath = screenshotTest.m_localBaselineScreenshotFilePath;
+                    AZStd::string screenshotImagePath = screenshotTest.m_screenshotFilePath;
+                    ImageComparisonToleranceLevel toleranceLevel = screenshotTest.m_toleranceLevel;
+                    AZStd::string toleranceLevelString = toleranceLevel.ToString();
+                    ImageComparisonResult localComparison = screenshotTest.m_localComparisonResult;
+                    AZStd::string localComparisonString = localComparison.GetSummaryString();
+                    ImageComparisonResult officialComparison = screenshotTest.m_officialComparisonResult;
+                    AZStd::string officialComparisonString = officialComparison.GetSummaryString();
```

Comment:
I'm assuming these variables are just for readability on the subsequent lines. You could go without the variables altogether. But it's fine either way. One optimization to avoid copying all of these strings and other types unnecessarily is to use const AZStd::string& or const WhateverType& or const auto&.

const AZStd::string& baselineImagePath = screenshotTest.m_localBaselineScreenshotFilePath;
const AZStd::string& screenshotImagePath = screenshotTest.m_screenshotFilePath;
const ImageComparisonToleranceLevel& toleranceLevel = screenshotTest.m_toleranceLevel;
...

It looks like you can do that for every variable in this loop which will make it a lot faster by not copying large string buffers for all of these variables if you're outputting a lot of data.